### PR TITLE
Use a themable WebDAV path all over.

### PR DESCRIPTION
**Pull request by dragotin**
This is needed for enterprise#481

@danimo @ogoffart please review.

### CODE REVIEW

**Comment by ogoffart** [comment]
I would like to avoid too much coupling between libsync and the theme.
In libsync, please use the path from the Account class rather than the one from the Theme.

File: src/libsync/owncloudpropagator.cpp [hunk 1]
```diff
@@ -341,7 +343,7 @@ void OwncloudPropagator::start(const SyncFileItemVector& items)
 bool OwncloudPropagator::isInSharedDirectory(const QString& file)
 {
     bool re = false;
-    if( _remoteDir.contains("remote.php/webdav/Shared") ) {
+    if( _remoteDir.contains( Theme::instance()->webDavPath() + QLatin1String("Shared") ) ) {
```
Comment (ogoffart) [inline src/libsync/owncloudpropagator.cpp]: Please use the webdav path from the account instead of from the theme here.

**Comment by dragotin** [comment]
@ogoffart: see my last patch

**Comment by ogoffart** [comment]
:+1:

**Comment by dragotin** [comment]
superseeded by https://github.com/owncloud/client/pull/3802

**Pull request closed without merging**